pkg/transaction/transaction.go

```diff
@@ -42,13 +41,20 @@ type Transaction struct {
 	ID     int
 	Path   string
 	Merges map[string]*Merge
+
 	status transactionState
 }
 
 type Interface interface {
-	Init(deployment.Deployment) error
+	Init(deployment.Deployment) (UpgradeHelper, error)
```

Comment:
This makes explicit the coupling between the UpgradeHelper and Transaction interfaces.